controllers/workspace/provision/deployment.go

```diff
@@ -107,6 +109,35 @@ func SyncDeploymentToCluster(
 		}
 	}
 
+	// Get pod and update dw status on crashloopbackoff
+	podList, err := getPods(workspace, clusterAPI.Client)
+	if err != nil {
+		return DeploymentProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{
+				Requeue: true,
+				Err:     err,
+			},
+		}
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodPending {
+			return DeploymentProvisioningStatus{
+				ProvisioningStatus: ProvisioningStatus{Requeue: true},
+			}
+		}
+
+		for _, containerStatuses := range pod.Status.ContainerStatuses {
+			if containerStatuses.State.Waiting != nil && containerStatuses.State.Waiting.Reason == constants.ContainerCrashLoopBackOffReason {
+				return DeploymentProvisioningStatus{
+					ProvisioningStatus: ProvisioningStatus{
+						FailStartup: true,
```

Comment:
let's provide a message with a failure cause here, `some of devworkspace containers crashed`

Reply:
We can even propagate the specific reason: `Container <containerName> has state CrashLoopBackOff`

Reply:
Good idea! I've updated to add a message!